datashader/transfer_functions/__init__.py

```diff
@@ -375,8 +346,10 @@ def _colorize(agg, color_key, how, span, min_alpha, name):
         # i.e. a 0 will be fully transparent, but any non-zero number will
         # be clipped to the span range and have min-alpha applied
         offset = np.array(span, dtype=data.dtype)[0]
-        if offset <= 0:
+        if np.nanmin(total) == 0 and total.dtype.kind == 'u':
```

Comment:
If the data is of `uint` type we should **always** mask out zeros even with a specified span, otherwise zero values end up being black in the aggregate which is wrong.